src/ReverseProxy/Forwarder/FastCgiHttpMessageHandler.cs

```diff
@@ -0,0 +1,1131 @@
+        public override void SetLength(long value) => stream.SetLength(value);
+        public override void Write(byte[] buffer, int offset, int count) => stream.Write(buffer, offset, count);
+
+        public override bool CanRead => stream.CanRead && !_disposed;
+        public override bool CanSeek => stream.CanSeek;
+        public override bool CanWrite => stream.CanWrite;
+        public override long Length => stream.Length;
+        public override long Position { get => throw new NotSupportedException(); set => throw new NotSupportedException(); }
+    }
+
+    private sealed class FastCgiStdinStream(ConnectionContext connection) : Stream
+    {
+        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        {
+            var count = buffer.Length;
+            var offset = 0;
+
+            while (count > 0)
+            {
+
+                var written = count < FastCgiRecordHeader.MAX_CONTENT_SIZE ? count : FastCgiRecordHeader.MAX_CONTENT_SIZE;
+
+                FastCgiHttpMessageHandler.Send(new FastCgiRecord(
+                    Header: new FastCgiRecordHeader(Type: FastCgiRecordHeader.RecordType.Stdin, ContentLength: (ushort)written),
+                    ContentData: new ReadOnlySequence<byte>(buffer.Slice(offset, written))), connection);
+
+                count -= written;
+                offset += written;
+            }
+
+            return ValueTask.CompletedTask;
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            while (count > 0)
+            {
+
+                var written = count < FastCgiRecordHeader.MAX_CONTENT_SIZE ? count : FastCgiRecordHeader.MAX_CONTENT_SIZE;
+
+                FastCgiHttpMessageHandler.Send(
+                    new FastCgiRecordHeader(Type: FastCgiRecordHeader.RecordType.Stdin, ContentLength: (ushort)written),
+                    buffer.AsSpan(offset, written), connection);
+
+                count -= written;
+                offset += written;
+            }
+        }
+
+        public override bool CanRead { get; } = false;
+        public override bool CanSeek { get; } = false;
+        public override bool CanWrite { get; } = true;
+        public override long Length { get; } = 0;
+
+        public override long Position { get; set; }
+
+        public override async Task FlushAsync(CancellationToken cancellationToken)
+        {
+            await FastCgiHttpMessageHandler.FlushAsync(connection, cancellationToken);
+        }
+
+        public override void Flush() => throw new NotSupportedException();
+        public override int Read(byte[] buffer, int offset, int count) => throw new NotSupportedException();
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+        public override void SetLength(long value) => throw new NotSupportedException();
+    }
+
+    private static class HttpResponseHeaders
```

Comment:
HttpResponseHeaders already exists
```suggestion
    private static class FastCgiResponseHeaders
```